src/js/firstDiscoveryEditor.js

```diff
@@ -37,11 +40,72 @@ https://github.com/fluid-project/infusion/raw/master/Infusion-LICENSE.txt
                         }
                     }
                 }
+            },
+            navButtons: {
+                type: "gpii.firstDiscovery.navButtons",
+                container: "{that}.dom.navButtons",
+                createOnEvent: "onCreateNavButtons",
+                options: {
+                    gradeNames: ["fluid.prefs.msgLookup"],
+                    members: {
+                        messageResolver: "{firstDiscoveryEditor}.msgResolver"
+                    },
+                    strings: {
+                        back: "{that}.msgLookup.back",
+                        next: "{that}.msgLookup.next",
+                        start: "{that}.msgLookup.start",
+                        finish: "{that}.msgLookup.finish"
+                    },
+                    panelTotalNum: "{firstDiscoveryEditor}.panelTotal",
+                    model: {
+                        currentPanelNum: "{firstDiscoveryEditor}.model.currentPanelNum"
+                    }
+                }
+            }
+        },
+        model: {
+            currentPanelNum: 1
+        },
+        modelListeners: {
+            "currentPanelNum": {
+                listener: "{that}.showPanel",
+                excludeSource: "init"
             }
         },
         events: {
-            onPrefsEditorReady: null
+            onPrefsEditorReady: null,
+            onCreateNavButtons: null
+        },
+        listeners: {
+            "onPrefsEditorReady.getPanelTotal": {
+                listener: "gpii.firstDiscovery.getPanelTotal",
+                args: ["{that}"],
+                priority: "first"
+            },
+            "onPrefsEditorReady.showInitialPanel": {
+                listener: "{that}.showPanel",
+                args: ["{that}.model.currentPanelNum"]
```

Comment:
i don't think this arg is necessary.  It looks like it is passed in from the invoker.